src/AutorestSwift/View Models/KeyValueViewModel.swift

```diff
@@ -185,26 +155,84 @@ struct KeyValueViewModel: Comparable {
+             .boolean,
+             .number:
             return "String(\(value))"
         // For these types, a variable will be created in the method using the naming convention `{key|value}String`
         case .date,
              .unixTime,
              .dateTime,
              .byteArray:
             return "\(value)String"
-        case .number:
-            return signatureParameter.required ? "\(value)String" : "String(\(value))"
         case .choice,
              .sealedChoice:
             return "\(value).rawValue"
         case .array:
             return "\(value).map { String($0) }.joined(separator: \"\(signatureParameter.delimiter)\") "
+        case .duration:
+            return "DateComponentsFormatter().string(from: \(value)) ?? \"\""
         default:
             return "\(value)"
         }
     }
 
+    private static func getValue(for constantSchema: ConstantSchema, isSkipUrlEncoding: Bool) -> String {
+        let constantValue: String = constantSchema.value.value
+        let type = constantSchema.valueType.type
+
+        if type == .string,
+            isSkipUrlEncoding {
+            return "\"\(constantValue)\".removingPercentEncoding ?? \"\""
+        } else {
+            switch type {
+            case .date,
+                 .unixTime,
+                 .dateTime,
+                 .byteArray,
+                 .string:
+                return "\"\(constantValue)\""
+            case .number:
+                let numberSchema = constantSchema.valueType
+                let swiftType = numberSchema.swiftType()
+                if swiftType == "Decimal" {
+                    return "\(swiftType)(\(constantValue))"
+                } else {
+                    return "String(\(swiftType)(\(constantValue)))"
+                }
+            default:
+                return "String(\(constantValue))"
+            }
+        }
+    }
+
+    private static func getKeyValueDecodeStrategy(for parameter: ParameterType) -> KeyValueDecodeStrategy {
+        let type = parameter.schema.type
+        // if parameter is from method signature (not from option) and type is date or byteArray,
+        // add decoding logic to string in the method and specify the right decoding strategy
+        switch type {
+        case .date,
+             .unixTime:
+            return .date
+        case .dateTime:
+            return .dateTime
+        case .byteArray:
+            if let byteArraySchema = parameter.schema as? ByteArraySchema,
+                byteArraySchema.format == .base64url {
+                return .base64ByteArray
+            } else {
+                return .byteArray
+            }
+        case .number:
+            if let numberSchema = parameter.schema as? NumberSchema {
+                return (numberSchema.swiftType() == "Decimal") ? .decimal : .number
+            } else {
+                return .number
+            }
+        default:
+            return .default
+        }
+    }
```

Comment:
Is there a reason this can't just be a computed property directly on `ParameterType`?

Reply:
Good suggestion. Have moved the code to `ParameterType`